### PR TITLE
Update package name

### PR DESCRIPTION
**Pull request by kirek007**
This PR consist changes related to renaming package and group for code and artifacts from `com.github.containersolutions` to `io.javaoperatorsdk`.

Changes includes:
✔ Java source code
✔ Readme file
✔ Maven configuration files
✔ Manifest files
✔ Site files

### CODE REVIEW

File: operator-framework/src/test/java/io/javaoperatorsdk/operator/ConcurrencyIT.java [hunk 1]
```diff
@@ -48,13 +47,13 @@ public void manyResourcesGetCreatedUpdatedAndDeleted() {
             log.info("Creating {} new resources", NUMBER_OF_RESOURCES_CREATED);
             for (int i = 0; i < NUMBER_OF_RESOURCES_CREATED; i++) {
                 TestCustomResource tcr = integrationTest.createTestCustomResource(String.valueOf(i));
-                integrationTest.getCrOperations().inNamespace(TEST_NAMESPACE).create(tcr);
```
Comment (kirek007) [inline operator-framework/src/test/java/io/javaoperatorsdk/operator/ConcurrencyIT.java:51]: This files consist undesired changes related to auto refactoring code using IDEA, but it looks good.

File: operator-framework/src/test/java/io/javaoperatorsdk/operator/ControllerExecutionIT.java [hunk 2]
```diff
@@ -38,7 +37,7 @@ public void configMapGetsCreatedForTestCustomResource() {
         integrationTestSupport.teardownIfSuccess(() -> {
             TestCustomResource resource = testCustomResource();
 
-            integrationTestSupport.getCrOperations().inNamespace(TEST_NAMESPACE).create(resource);
```
Comment (kirek007) [inline operator-framework/src/test/java/io/javaoperatorsdk/operator/ControllerExecutionIT.java:41]: This files consist undesired changes related to auto refactoring code using IDEA, but it looks good.

File: operator-framework/src/test/java/io/javaoperatorsdk/operator/ControllerUtilsTest.java [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
-import static com.github.containersolutions.operator.api.Controller.DEFAULT_FINALIZER;
-import static com.github.containersolutions.operator.sample.TestCustomResourceController.CRD_NAME;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 class ControllerUtilsTest {
 
     @Test
     public void returnsValuesFromControllerAnnotationFinalizer() {
-        assertEquals(DEFAULT_FINALIZER, ControllerUtils.getDefaultFinalizer(new TestCustomResourceController(null)));
```
Comment (kirek007) [inline operator-framework/src/test/java/io/javaoperatorsdk/operator/ControllerUtilsTest.java:17]: This files consist undesired changes related to auto refactoring code using IDEA, but it looks good.

File: operator-framework/src/test/java/io/javaoperatorsdk/operator/processing/retry/GenericRetryExecutionTest.java [hunk 4]
```diff
@@ -20,13 +18,13 @@ public void delayIsMultipliedEveryNextDelayCall() {
         RetryExecution retryExecution = getDefaultRetryExecution();
 
         Optional<Long> res = callNextDelayNTimes(retryExecution, 2);
-        assertThat(res.get()).isEqualTo(DEFAULT_INITIAL_INTERVAL);
```
Comment (kirek007) [inline operator-framework/src/test/java/io/javaoperatorsdk/operator/processing/retry/GenericRetryExecutionTest.java:23]: This files consist undesired changes related to auto refactoring code using IDEA, but it looks good.

**Comment by csviri** [comment]
@adam-sandor this won't break the release to maven central? Aren't we registered there to some package name?

**Comment by adam-sandor** [comment]
> @adam-sandor this won't break the release to maven central? Aren't we registered there to some package name?
> Also I think "github" needs to be in the package @kirek007

Yes that is right. There were two options to register ourselves in MC the one that looked simpler was Github. I think for the other you actually have to own the domain (but I don't really remember this). We will definitely have to redo the MC registration process.

**Comment by kirek007** [comment]
> @adam-sandor this won't break the release to maven central? Aren't we registered there to some package name?
> Also I think "github" needs to be in the package @kirek007
this change WILL BREAK pushing to maven central, but we should also update that part and start pushing to new package there.

**Pull request merged**